pkg/kubelet/util/boottime_util_linux.go

```diff
@@ -21,17 +21,49 @@ package util
 
 import (
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
 	"time"
 
 	"golang.org/x/sys/unix"
+	"k8s.io/klog/v2"
 )
 
-// GetBootTime returns the time at which the machine was started, truncated to the nearest second
+// GetBootTime returns the time at which the machine was started, truncated to the nearest second.
+// It uses /proc/uptime first, which is more accurate, and falls back to the less accurate
+// unix.Sysinfo if /proc/uptime failed.
 func GetBootTime() (time.Time, error) {
+	bootTime, err := getBootTimeWithProcUptime()
+	if err != nil {
+		klog.InfoS("Failed to get boot time from /proc/uptime. Will retry with unix.Sysinfo.", "error", err)
+		return getBootTimeWithSysinfo()
+	}
+	return bootTime, nil
+}
+
+func getBootTimeWithProcUptime() (time.Time, error) {
+	currentTime := time.Now()
+	raw, err := os.ReadFile("/proc/uptime")
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error getting system uptime: %w", err)
+	}
+	rawFields := strings.Fields(string(raw))
+	if len(rawFields) != 2 {
+		return time.Time{}, fmt.Errorf("unexpected value from /proc/uptime: %s", raw)
+	}
+	uptime, err := strconv.ParseFloat(rawFields[0], 64)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("error parsing uptime %s: %w", rawFields[0], err)
+	}
+	return currentTime.Add(-time.Duration(uptime * float64(time.Second))).Truncate(time.Second), nil
```

Comment:
Would `Round` be more reliable than `Truncate` here?
```suggestion
	return currentTime.Add(-time.Duration(uptime * float64(time.Second))).Round(time.Second), nil
```

Reply:
I think they are the same in terms of reliability and consistency:  for `round`, the corner case happens around half seconds; for `truncate`, the corner case happens around whole seconds. If the true timestamps follow even distribution, then the chances of corner case happening should be the same.

I kept truncate because was the original behavior (cc @dashpole ), so node start time won't drift when upgrading to this new version of kubelet.

Reply:
Updated the change to use `/proc/stat` so this is no longer applicable.

B.T.W. the linue kernel calculation of boot time is [returning the second and discarding subsecond](https://github.com/torvalds/linux/blob/v6.11/fs/proc/stat.c#L175-L180), equivalent to our `truncate` here.